### PR TITLE
[RW-4314][risk=no] Gauge metric for GSuite Domain Account Count

### PR DESCRIPTION
**Pull request by jaycarlton**
The closest API I could find for this information was the [Users List API](https://developers.google.com/admin-sdk/directory/v1/reference/users/list). There's no count or summary I could find, so I'm hitting it with the max page size of 500 and then fetching more. I might be able to reduce the size of each user returned using the fields attribute.

I put a label on the gauge for the domain name. This is both a reminder and for future-proofness, in case we wanted to have multiple domains per environment or somehow look at multiple environments in the same chart someday. We don't expect either of those to happen yet.

Here's a screen shot with it in action.
Screen Shot 2020-01-22 at 11 56 14 AM

[Stackdriver dashboard link](https://console.cloud.google.com/monitoring/dashboards/custom/18329516758933998444?project=all-of-us-workbench-test&timeDomain=1d).
---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java [hunk 1]
```diff
@@ -239,6 +252,38 @@ public void deleteUser(String username) {
...
   }
 
+  @Override
+  public Collection<MeasurementBundle> getGaugeData() {
+    long userCount = 0;
+    try {
+      final Directory directoryService = getGoogleDirectoryService();
+      Optional<String> nextPageToken = Optional.empty();
+      do {
+        final Directory.Users.List listQuery = directoryService
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java]: I don't know of a good way to write a reasonable test case for this logic

File: api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java [hunk 2]
```diff
@@ -239,6 +252,40 @@ public void deleteUser(String username) {
...
+                .list()
+                .setDomain(gSuiteDomain())
+                .setViewType(USER_VIEW_TYPE)
+                .setMaxResults(MAX_USERS_LIST_PAGE_SIZE)
+                .setOrderBy(EMAIL_USER_FIELD);
+        nextPageToken.ifPresent(listQuery::setPageToken);
+
+        final Users usersQueryResult = listQuery.execute();
+
+        userCount += usersQueryResult.getUsers().size();
```
Comment (jaycarlton) [inline api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java]: Do we have a place in the GSuite Admin UI where I can verify this value? Looks like 660 for fake-research-aou.org.

**Comment by jaycarlton** [comment]
@gjuggler this looks independent from #3016 but just a heads up. I'd also like to see if you know of a way to simply get the user count without stepping through every page.

File: api/src/main/java/org/pmiops/workbench/monitoring/attachments/MetricLabel.java [hunk 3]
```diff
@@ -10,9 +10,12 @@
 public enum MetricLabel implements MetricLabelBase {
   BUFFER_ENTRY_STATUS("BufferEntryStatus", Enums.getValueStrings(BufferEntryStatus.class)),
   DATASET_INVALID("Invalid", Booleans.VALUE_STRINGS),
+  DATA_ACCESS_LEVEL("DataAccessLevel", Enums.getValueStrings(DataAccessLevel.class)),
+  GSUITE_DOMAIN(
+      "gsuite_domain"), // provide this (slightly redundant) in case we ever merge time series
```
Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/monitoring/attachments/MetricLabel.java]: the formatter butchered this a bit, probably move the comment before the enum value instead

File: api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java [hunk 4]
```diff
@@ -239,6 +252,40 @@ public void deleteUser(String username) {
...
+    long userCount = 0;
+    try {
+      final Directory directoryService = getGoogleDirectoryService();
+      Optional<String> nextPageToken = Optional.empty();
+      do {
+        final Directory.Users.List listQuery =
+            directoryService
+                .users()
+                .list()
+                .setDomain(gSuiteDomain())
```
[the remark this answers is not in the document]
Comment (jaycarlton) [reply api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java]: IIUC, we are on the right track. The gauge will fire for all environments, and we'll be able to distinguish those based on the `namespace` and the `project_id` in the resource descriptor.

Where it gets tricky is that those environments today are in separate [Stackdriver Workspaces](https://cloud.google.com/monitoring/workspaces/), meaning we can't see all those counts in one place to apply some kind of formula. I'm quietly pushing for a bit more investigation of moving things to a single workspace, but the catch is that no project can be in more than one workspace at a time.

[the remark this answers is not in the document]
Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java]: If that's not possible today, then I'd suggest just dropping this filter and have the metric represent "the total number of users in the corresponding gsuite domain". The metric would be identical in all environments which share a gsuite instance, and would be larger than the actual user count for the environment.

File: api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java [hunk 5]
```diff
@@ -239,6 +252,40 @@ public void deleteUser(String username) {
...
+                .setViewType(USER_VIEW_TYPE)
```
[the remark this answers is not in the document]
Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java]: Ah, I see that it doesn't apply to the basic fields - misunderstood the docs for this. Yeah - sounds fine.

**Pull request merged**